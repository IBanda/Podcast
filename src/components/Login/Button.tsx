import React from 'react';
import { TouchableHighlight, View, StyleSheet, Text } from 'react-native';

interface Props {
  children: string;
  onPress: () => void;
  bg: string;
}

export default function Button({ children, onPress, bg }: Props) {
  return (
    <TouchableHighlight
      onPress={onPress}
      style={[styles.touchableHighlight, { shadowColor: bg }]}
    >
      <View style={[styles.button, { backgroundColor: bg, shadowColor: bg }]}>
        <Text style={styles.buttonText}>{children}</Text>
      </View>
    </TouchableHighlight>
  );
}

const styles = StyleSheet.create({
  touchableHighlight: {
    shadowOffset: {
      width: 0,
      height: -5,
    },
    shadowOpacity: 0.44,
    shadowRadius: 20,
    borderRadius: 25,
    marginBottom: 20,
  },
  button: {
    borderRadius: 25,
    alignItems: 'center',
    paddingVertical: 10,
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.2,
    shadowRadius: 20,
    elevation: 16,
  },
  buttonText: {
    color: '#fff',
  },
});
