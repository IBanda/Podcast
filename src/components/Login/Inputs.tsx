import React, { useState } from 'react';
import { View, TextInput, Image, StyleSheet } from 'react-native';

interface Prop {
  placeholder: string;
  type: 'emailAddress' | 'password';
  value: string;
  onChangeText: (text: string) => void;
  children: object;
}

function Input({ placeholder, type, value, onChangeText, children }: Prop) {
  return (
    <View style={styles.inputContainer}>
      {children}
      <TextInput
        style={styles.input}
        placeholder={placeholder}
        textContentType={type}
        value={value}
        onChangeText={onChangeText}
      />
    </View>
  );
}

export default function LoginInputs() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const onEmailChange = (text: string) => {
    setEmail(text);
  };
  const onPasswordChange = (text: string) => {
    setPassword(text);
  };
  return (
    <View style={styles.container}>
      <Input
        placeholder="E-mail address"
        type="emailAddress"
        onChangeText={onEmailChange}
        value={email}
      >
        <Image
          source={require('../../../assets/login/email.png')}
          style={styles.inputIcon}
        />
      </Input>
      <Input
        placeholder="Password"
        type="password"
        onChangeText={onPasswordChange}
        value={password}
      >
        <Image
          source={require('../../../assets/login/key.png')}
          style={styles.inputIcon}
        />
      </Input>
    </View>
  );
}

const styles = StyleSheet.create({
  inputContainer: {
    borderStyle: 'solid',
    borderWidth: 1,
    borderColor: '#fff',
    borderRadius: 10,
    borderBottomRightRadius: 0,
    marginBottom: 10,
    flexDirection: 'row',
    alignItems: 'center',
    position: 'relative',
    overflow: 'hidden',
  },
  inputIcon: {
    width: 15,
    height: 15,
    marginRight: 5,
    position: 'absolute',
    left: 15,
  },
  input: {
    width: '100%',
    height: '100%',
    padding: 15,
    paddingLeft: 40,
    color: '#fff',
  },
  container: {
    justifyContent: 'space-between',
    marginBottom: 20,
  },
});
