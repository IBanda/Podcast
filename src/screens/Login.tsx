import React from 'react';
import {
  ImageBackground,
  StyleSheet,
  Text,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import Button from '../components/Login/Button';
import LoginInputs from '../components/Login/Inputs';
import Logo from '../components/Logo';

const bgImage = require('../../assets/login/bg.png');

export default function Login() {
  return (
    <View style={styles.container}>
      <ImageBackground source={bgImage} style={styles.imageBackground}>
        <View style={[styles.container, styles.layer]}>
          <Logo />
          <Text style={styles.text}>Episodic series of digital audio</Text>
          <LoginInputs />
          <Button onPress={() => {}} bg="#3369FF">
            Login
          </Button>
          <TouchableWithoutFeedback>
            <Text style={styles.forgottenPWDText}>Forgot Password?</Text>
          </TouchableWithoutFeedback>
          <Button onPress={() => {}} bg="#2B5D95">
            Login with Facebook
          </Button>
          <Button onPress={() => {}} bg="#FF334B">
            Register new account
          </Button>
        </View>
      </ImageBackground>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  imageBackground: {
    flex: 1,
    resizeMode: 'cover',
    paddingRight: 30,
    paddingBottom: 50,
  },
  layer: {
    borderBottomRightRadius: 20,
    overflow: 'hidden',
    backgroundColor: 'black',
    paddingHorizontal: 30,
    paddingVertical: 30,
  },
  text: {
    color: '#fff',
    fontSize: 20,
    marginVertical: 20,
    maxWidth: '70%',
  },
  forgottenPWDText: {
    color: '#fff',
    margin: 'auto',
    marginBottom: 20,
    marginTop: 0,
    textDecorationLine: 'underline',
  },
});
