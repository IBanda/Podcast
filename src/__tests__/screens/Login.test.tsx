import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../screens/Login';

it('renders the login screen', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
