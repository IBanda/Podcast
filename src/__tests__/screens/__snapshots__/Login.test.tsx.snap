// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the login screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "paddingBottom": 50,
        "paddingRight": 30,
        "resizeMode": "cover",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "black",
            "borderBottomRightRadius": 20,
            "overflow": "hidden",
            "paddingHorizontal": 30,
            "paddingVertical": 30,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 72,
            "width": 156,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "marginVertical": 20,
            "maxWidth": "70%",
          }
        }
      >
        Episodic series of digital audio
      </Text>
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "marginBottom": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomRightRadius": 0,
              "borderColor": "#fff",
              "borderRadius": 10,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 15,
                "left": 15,
                "marginRight": 5,
                "position": "absolute",
                "width": 15,
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="E-mail address"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#fff",
                "height": "100%",
                "padding": 15,
                "paddingLeft": 40,
                "width": "100%",
              }
            }
            textContentType="emailAddress"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomRightRadius": 0,
              "borderColor": "#fff",
              "borderRadius": 10,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 15,
                "left": 15,
                "marginRight": 5,
                "position": "absolute",
                "width": 15,
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Password"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#fff",
                "height": "100%",
                "padding": 15,
                "paddingLeft": 40,
                "width": "100%",
              }
            }
            textContentType="password"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 25,
              "marginBottom": 20,
              "shadowOffset": Object {
                "height": -5,
                "width": 0,
              },
              "shadowOpacity": 0.44,
              "shadowRadius": 20,
            },
            Object {
              "shadowColor": "#3369FF",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "elevation": 16,
                "paddingVertical": 10,
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 20,
              },
              Object {
                "backgroundColor": "#3369FF",
                "shadowColor": "#3369FF",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Login
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "color": "#fff",
            "margin": "auto",
            "marginBottom": 20,
            "marginTop": 0,
            "textDecorationLine": "underline",
          }
        }
      >
        Forgot Password?
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 25,
              "marginBottom": 20,
              "shadowOffset": Object {
                "height": -5,
                "width": 0,
              },
              "shadowOpacity": 0.44,
              "shadowRadius": 20,
            },
            Object {
              "shadowColor": "#2B5D95",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "elevation": 16,
                "paddingVertical": 10,
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 20,
              },
              Object {
                "backgroundColor": "#2B5D95",
                "shadowColor": "#2B5D95",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Login with Facebook
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 25,
              "marginBottom": 20,
              "shadowOffset": Object {
                "height": -5,
                "width": 0,
              },
              "shadowOpacity": 0.44,
              "shadowRadius": 20,
            },
            Object {
              "shadowColor": "#FF334B",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "elevation": 16,
                "paddingVertical": 10,
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 20,
              },
              Object {
                "backgroundColor": "#FF334B",
                "shadowColor": "#FF334B",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Register new account
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
